src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6005,6 +6005,7 @@ private void VisitConditionalAccess(BoundConditionalAccess node, out PossiblyCon
 
             // Per LDM 2019-02-13 decision, the result of a conditional access "may be null" even if
             // both the receiver and right-hand-side are believed not to be null.
+            UnsplitIfNeeded(resultType);
```

Comment:
I thought if we are analyzing `"a"?.M1(x)`, and `M1` returns `bool`, then `resultType` is `bool?` here. So why would we ever keep in a split state at this point? Would we achieve the correct behavior more simply if we just always called Unsplit() after the `VisitPossibleConditionalAccess` on L5928 (the non-null-const receiver case)?

Reply:
Sounds good, let me try that. Thanks.

Reply:
Turns out it is not easy to move `UnsplitIfNeeded` into the `if` above - we don't know the `resultType` there (the `bool?`) we only know `LvalueResultType` but that's `bool`. We would need to compute the `resultType` sooner, and in two places. Which seems unnecessary - we can just leave the `UnsplitIfNeeded` here where it is.

Reply:
Right, sorry, I misread your original suggestion, it makes sense, let me try that.